utils/cp_parallel.py

```diff
@@ -0,0 +1,141 @@
+"""
+This collection of functions runs CellProfiler in parallel and can convert the results into log files
+for each process.
+"""
+
+import multiprocessing
+import logging
+from typing import List
+import os
+import subprocess
+import pathlib
+from concurrent.futures import ProcessPoolExecutor, Future
+from errors.exceptions import MaxWorkerError
+
+
+def results_to_log(
+    results: List[subprocess.CompletedProcess], log_dir: pathlib.Path, run_name: str
+) -> None:
+    """
+    This function will take the list of subprocess.results from a CellProfiler parallelization run and
+    convert into a log file for each process.
+
+    Args:
+        results (List[subprocess.CompletedProcess]): the outputs from a subprocess.run
+        log_dir (pathlib.Path): directory for log files
+        run_name (str): a given name for the type of CellProfiler run being done on the plates (example: whole image features)
+    """
+    # Access the command (args) and stderr (output) for each CompletedProcess object
+    for result in results:
+        # assign plate name and decode the CellProfiler output to use in log file
+        plate_name = result.args[6].name
```

Comment:
Is there a more dynamic way to achieve this instead of relying on an index? Using a fixed index can lead to future bugs if the structure of args changes.

Reply:
This is a good point! It might be easier to assign it another way like have it assigned through the dictionary needed to run CellProfiler. Let me look into it and I will consider the change. Thanks!